carma_wm_ctrl/src/WMBroadcaster.cpp

```diff
@@ -2075,6 +2078,60 @@ lanelet::Lanelet  WMBroadcaster::createLinearInterpolatingLanelet(const lanelet:
   return lanelet::Lanelet(lanelet::utils::getId(), createLinearInterpolatingLinestring(left_front_pt, left_back_pt, increment_distance), createLinearInterpolatingLinestring(right_front_pt, right_back_pt, increment_distance));
 }
 
+void WMBroadcaster::updateUpcomingSGIntersectionIds(const lanelet::Lanelet cur_lanelet)
+{
+  uint16_t map_msg_intersection_id = 0;
+  uint16_t cur_signal_group_id = 0;
+  for(auto itr = sim_.signal_group_to_entry_lanelet_ids_.begin(); itr != sim_.signal_group_to_entry_lanelet_ids_.end(); itr++)
+  {
+    if(itr->second.find(cur_lanelet.id()) != itr->second.end())
+    {
+      cur_signal_group_id = itr->first;
+    }
+  }
```

Comment:
I think the idea is correct. Although it will only get signals if we are on the entry lanelet. Since we need the element way before entry lanelet, I suggest using `getSignalsAlongRoute()` to get the nearest traffic light. And use sim_ 's another mapping `signal_group_to_traffic_light_id_;` to get the signal group 


https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/carma_wm/include/carma_wm/CARMAWorldModel.h#L228